validator_client/src/initialized_validators.rs

```diff
@@ -648,6 +654,12 @@ impl InitializedValidators {
         self.validators.get(public_key).and_then(|v| v.gas_limit)
     }
 
+     /// Returns the `builder_registration_pubkey_override` for a given public key specified in the
+    /// `ValidatorDefinitions`.
+    pub fn builder_registration_pubkey_override(&self, public_key: &PublicKeyBytes) -> Option<PublicKey> {
```

Comment:
Thoughts were something like this function would need to be called in 

publish_validator_registration_data

using each pubkey from ValidatorRegistrationKey as input for public_key

not sure about the best way to get the InitializedValidator object into preparation_service.rs

Reply:
I modified the `collect_validator_registration_keys` to collect the override to try to minimize how often we're accessing the val defs file